index.js

```diff
@@ -95,7 +95,7 @@ function handleArgs(command, args, options = {}) {
 
 	if (process.platform === 'win32' && path.basename(command, '.exe') === 'cmd') {
 		// #116
-		args.unshift('/q');
+		args = [...new Set(['/q', '/d', '/s', '/c', ...args])];
```

Comment:
Why is this needed: does `cmd.exe` complain on duplicate CLI arguments?

Also this probably should be in a separate PR.

Reply:
`node-cross-spawn` add those options when in `shell` mode and when the command file is not a ".exe". Or fork method runs `cmd.exe` in non-shell mode. Without those options, the cmd script will never quit and pipe its output.
The order is important, that's why I added a non-duplicate safety here.

Reply:
I think you might have used the French "Or" there :)

You're speaking about [those lines](https://github.com/moxystudio/node-cross-spawn/blob/48bee1dd5cf06f8162cbbce812b5c6e01b1c439d/lib/parse.js#L45), right? Also just for reference those lines just emulate what [Node.js does with `shell: true`](https://github.com/nodejs/node/blob/6f2bd9dcc3a9ed6cf509cdcd381c1c456f0bd3cb/lib/child_process.js#L483).

I might be wrong here, but it seems to me that when `shell: false` (like for `fork()`):
  - if the command is `*.exe` or `*.cmd`, `node-cross-spawn` does not touch it.
  - otherwise, it calls the command with `cmd.exe /d /s /c` (like `shell: true` would).

Here the `fork` method will run `node` (not `cmd.exe`) in non-shell mode, so `node-cross-spawn` should not touch it. Or am I missing something?

Reply:
> I think you might have used the French "Or" there :)

I'm loosing my mind apparently -__-

> You're speaking about those lines

Yes. When `fork` runs `cmd.exe` through the `execPath` option, it should adds those 3 options too. I'm still not sure if `fork` is intended to run any other executable than `node`.

Reply:
As far as I understand it, `fork` is not meant to run `cmd.exe` through the `execPath` option.

Now would the problem you're describing still happen when doing `execa('cmd.exe', ['file.exe'], { shell: false })` (which is what `fork` would eventually call)?

Reply:
I just tried and yes, the example you provide fail on Windows. The script never quit itself and block the thread.

Reply:
This would be a bug in the current code, not related to this PR (since it can be reproduced without `fork()`). Do you think you could open a separate issue about this, with a reproducible example? I could take a stab at fixing it.

Reply:
https://github.com/sindresorhus/execa/issues/232